packages/headless/src/ssr-next/common/controller-utils.ts

```diff
@@ -59,20 +61,32 @@ export function buildControllerDefinitions<
   ) as InferControllersMapFromDefinition<TControllerDefinitionsMap>;
 }
 
-export function createStaticState<TSearchAction extends UnknownAction>({
+export function createStaticState<
+  TSearchAction extends UnknownAction,
+  TControllerDefinitions extends ControllerDefinitionsMap<
+    CoreEngine | CoreEngineNext,
+    Controller
+  >,
+>({
   searchAction,
   controllers,
 }: {
   searchAction: TSearchAction;
-  controllers: ControllersMap;
+  controllers: InferControllersMapFromDefinition<TControllerDefinitions>;
 }): EngineStaticState<
   TSearchAction,
-  InferControllerStaticStateMapFromControllers<ControllersMap>
+  InferControllerStaticStateMapFromDefinitions<TControllerDefinitions>
 > {
   return {
-    controllers: mapObject(controllers, (controller) => ({
-      state: clone(controller.state),
-    })) as InferControllerStaticStateMapFromControllers<ControllersMap>,
+    controllers: mapObject(controllers, (controller) => {
+      const base = {state: clone(controller.state)};
+      // TODO: check if initialState is an object
+      if ('initialState' in controller) {
+        // TODO: check if that works!!!
+        return {...base, initialState: clone(controller.initialState)};
+      }
+      return base;
+    }) as InferControllerStaticStateMapFromDefinitions<TControllerDefinitions>,
```

Comment:
TODO: add a warning on the initialState prop for users to not access it. it is meant to be used by hydrateStaticState only